src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -175,14 +190,40 @@ const SignOrExecuteForm = ({
     return id
   }
 
+  const onRelay = async () => {
+    const [, createdTx] = assertDependencies()
+
+    const { gasLimit } = getTxOptions(advancedParams, currentChain)
+
+    let id = txId
+    let safeTx = createdTx
+    // Add missing signature
+    if (createdTx.signatures.size < safe.threshold) {
```

Comment:
Can you extract this logic to a function along the lines of `isRelayable` as the same logic is use in `useWalletCanRelay`. It could live alongside `isSignableBy`.

Reply:
Sure, it can be abstracted. However I'd call this condition something like `hasEnoughSignatures` like I've done in `useWalletCanRelay` WDYT?
I also think it does not quite fit the `transaction-guards` file, I'd rather move it to `utils/transactions` no?

Reply:
`hasEnoughSignatures` sounds fine by me. I'm impartial where you situate it.

Reply:
Done